src/common/datetime/calc_date.ts

```diff
@@ -35,5 +35,12 @@ export const calcDateProperty = (
   options?
 ) =>
   locale.time_zone === TimeZone.server
-    ? (calcZonedDate(date, config.time_zone, fn, options) as number | boolean)
+    ? (calcZonedDate(
+        date,
+        config.time_zone,
+        fn,
+        options instanceof Date
+          ? utcToZonedTime(options, config.time_zone)
+          : options
```

Comment:
This will fix it for this case, but since options can really be anything (we just pass it to the function provided) I think the caller of `calcDateProperty` should be responsible for this?

Reply:
Maybe just duplicate this function into a `calcDateDifferenceProperty` function with better typing would help?

I don't think I want to duplicate this at every calling instance because it happens a lot of places, and I think it's a bit esoteric how this works and would be better to keep it contained in one place. 

While it's true options can be anything, I would expect that for all the date functions if you pass two dates to the function, it could be assumed that you want to apply the same timezone shift to both (difference calculations are the only thing we use this function for).

Reply:
Yes, a second function taking 2 date params for difference functions sounds like a good idea.

Reply:
updated